app/containers/ChatContainer/selectors.js

```diff
@@ -0,0 +1,28 @@
+import { createSelector } from 'reselect';
+import { List } from 'immutable';
+
+/**
+ * Direct selector to the chatContainer state domain
+ */
+const selectChatContainerDomain = (state) => state.get('chatcontainer');
+
+/**
+ * Other specific selectors
+ */
+export const getMessages = createSelector(
+  selectChatContainerDomain,
+  (chatContainer) => chatContainer.get('messages', List()).toJS(),
```

Comment:
Read this as "in the chatContainer domain, get the messages key, defaulted to an empty Immutable List if it does not exist, and return it as a Javascript array"